src/meshes/p4est_mesh.jl

```diff
@@ -268,35 +300,118 @@ is non-trivial.
 """
 function P4estMesh{NDIMS}(meshfile::String;
-                   mapping=nothing, polydeg=1, RealT=Float64,
-                   initial_refinement_level=0, unsaved_changes=true) where NDIMS
-  # Prevent p4est from crashing Julia if the file doesn't exist
+                          mapping=nothing, polydeg=1, RealT=Float64,
+                          initial_refinement_level=0, unsaved_changes=true) where NDIMS
+  # Prevent `p4est` from crashing Julia if the file doesn't exist
   @assert isfile(meshfile)
 
-  conn = read_inp_p4est(meshfile, Val(NDIMS))
+  # Read in the Header of the meshfile to determine which constructor is approproate
+  header = open(meshfile, "r") do io
+    readline(io) # *Header of the Abaqus file; discarded
+    readline(io) # Readin the actual header information
+  end
+
+  # Check if the meshfile was generated using HOHQMesh
+  if header == " File created by HOHQMesh"
+    # Mesh curvature and boundary naming is handled with additional information available in meshfile
+    p4est, tree_node_coordinates, nodes, boundary_names = p4est_mesh_from_hohqmesh_abaqus(meshfile, initial_refinement_level,
+                                                                                          NDIMS, RealT)
+  else
+    # Mesh curvature is handled directly by applying the mapping keyword argument
+    p4est, tree_node_coordinates, nodes, boundary_names = p4est_mesh_from_standard_abaqus(meshfile, mapping, polydeg,
+                                                                                          initial_refinement_level,
+                                                                                          NDIMS, RealT)
+  end
+
+  return P4estMesh{NDIMS}(p4est, tree_node_coordinates, nodes,
+                          boundary_names, "", unsaved_changes)
+end
+
+
+# Create the mesh connectivity, mapped node coordinates within each tree, reference nodes in [-1,1]
+# and a list of boundary names for the `P4estMesh`. High-order boundary curve information as well as
+# the boundary names on each tree are provided by the `meshfile` created by
+# [`HOHQMesh.jl`](https://github.com/trixi-framework/HOHQMesh.jl).
+function p4est_mesh_from_hohqmesh_abaqus(meshfile, initial_refinement_level, n_dimensions, RealT)
+  # Create the mesh connectivity using `p4est`
+  conn = read_inp_p4est(meshfile, Val(n_dimensions))
+
+  # These need to be of the type Int for unsafe_wrap below to work
+  n_trees::Int = conn.num_trees
+  n_vertices::Int = conn.num_vertices
+
+  # Extract a copy of the element corners to compute the tree node coordinates
+  vertices = unsafe_wrap(Array, conn.vertices, (3, n_vertices))
+
+  # Readin all the information from the mesh file into a string array
+  file_lines = readlines(open(meshfile))
+
+  # Get the file index where the mesh polynomial degree is given in the meshfile
+  file_idx = findfirst(occursin.("** mesh polynomial degree", file_lines))
```

Comment:
```suggestion
  file_idx = findfirst(contains("** mesh polynomial degree"), file_lines)
```
Should be the same result but without creating a temporary array.

Reply:
This throws me an error
```julia
file_idx = findfirst(contains("** mesh polynomial degree", file_lines))
ERROR: MethodError: no method matching occursin(::Vector{String}, ::String)
Closest candidates are:
  occursin(::Union{AbstractChar, AbstractString}, ::AbstractString) at strings/search.jl:621
  occursin(::Regex, ::AbstractString; offset) at regex.jl:192
  occursin(::Any) at strings/search.jl:635
Stacktrace:
 [1] contains(haystack::String, needle::Vector{String})
   @ Base ./strings/util.jl:98
 [2] top-level scope
   @ REPL[16]:1
```

Reply:
Nevermind, I had a typo in the command. This works